lib/optimizely.rb

```diff
@@ -484,6 +510,71 @@ def get_feature_variable_double(feature_flag_key, variable_key, user_id, attribu
       variable_value
     end
 
+    # Get values of all the variables in the feature flag and returns them in a Dict
+    #
+    # @param feature_flag_key - String key of feature flag
+    # @param user_id - String user ID
+    # @param attributes - Hash representing visitor attributes and values which need to be recorded.
+    #
+    # @return [Dict] the Dict containing all the varible values
+    # @return [nil] if the feature flag is not found.
+
+    def get_all_feature_variables(feature_flag_key, user_id, attributes = nil)
+      unless is_valid
+        @logger.log(Logger::ERROR, InvalidProjectConfigError.new('get_all_feature_variables').message)
+        return nil
+      end
+
+      return nil unless Optimizely::Helpers::Validator.inputs_valid?(
+        {
+          feature_flag_key: feature_flag_key,
+          user_id: user_id
+        },
+        @logger, Logger::ERROR
+      )
+
+      return nil unless user_inputs_valid?(attributes)
+
+      config = project_config
+
+      feature_flag = config.get_feature_flag_from_key(feature_flag_key)
+      unless feature_flag
+        @logger.log(Logger::INFO, "No feature flag was found for key '#{feature_flag_key}'.")
+        return nil
+      end
+
+      decision = @decision_service.get_variation_for_feature(config, feature_flag, user_id, attributes)
+      variation = decision ? decision['variation'] : nil
+      feature_enabled = variation ? variation['featureEnabled'] : false
+      all_variables = {}
+
+      feature_flag['variables'].each do |variable|
+        variable_value = get_feature_variable_for_variation(feature_flag_key, feature_enabled, variation, variable, user_id)
+        all_variables[variable['key']] = Helpers::VariableType.cast_value_to_type(variable_value, variable['type'], @logger)
+      end
+
+      source_string = Optimizely::DecisionService::DECISION_SOURCES['ROLLOUT']
+      if decision && decision['source'] == Optimizely::DecisionService::DECISION_SOURCES['FEATURE_TEST']
+        source_info = {
```

Comment:
isn't this out of scope?

Reply:
No. It was decided to add a notification for all-feature-variables later. This is bein added in all the sdks now